react/features/chat/middleware.js

```diff
@@ -94,15 +95,18 @@ function _addChatMsgListener(conference, { dispatch, getState }) {
             // backfilled for a participant that has left the conference.
             const participant = getParticipantById(state, id) || {};
             const displayName = participant.name
-                || `${interfaceConfig.DEFAULT_REMOTE_DISPLAY_NAME} (${id})`;
+                || (typeof interfaceConfig === 'object'
+                && `${interfaceConfig.DEFAULT_REMOTE_DISPLAY_NAME} (${id})`);
             const hasRead = participant.local || isChatOpen;
 
-            APP.API.notifyReceivedChatMessage({
-                body: message,
-                id,
-                nick: displayName,
-                ts: timestamp
-            });
+            if (typeof APP !== 'undefined') {
+                APP.API.notifyReceivedChatMessage({
```

Comment:
Glad you brought that up. I agree. Right now a lot of the APP.API calls are scattered around the codebase. It would be great to consolidate those, maybe as a store change subscriber, but I think it would be better to put that on the tech debt backlog and nag me to do it.

Reply:
👍